code/src/Form.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react"
+
+import { Consumption } from "./Consumption"
+import { Number } from "./Number"
+import { Author } from "./Author"
+import { Summary } from "./Summary"
+
+
+
+
+export const Form = () => {
+    const [ consumption, setConsumption ] = useState("");
+    const [ numberOfBooks, setNumberOfBooks ] = useState("");
+    const [ favouriteAuthor, setFavouriteAuthor ] = useState("");
+    const [ showSummary, setShowSummary ] = useState(false);
+
+    const handleConsumptionChange = newConsumption => {
+        setConsumption(newConsumption)
+    }
+
+    const handleNumberOfBooksChange = newNumber => {
+        setNumberOfBooks(newNumber)
+    }
+
+    const handleFavouriteAuthorChange = newAuthor => {
+        setFavouriteAuthor(newAuthor)
+    }
+
+    const handleSubmit = event => {
+        event.preventDefault();
+        setShowSummary(true)
+    }
+
+    const handleRestart = () => {
+        setShowSummary(false);
+        setConsumption("");
+        setNumberOfBooks("");
+        setFavouriteAuthor("")
```

Comment:
good that you are making separate functions and pass in the function name instead of "setConsumption" :)